### PR TITLE
Update the wasi toolchain.

### PR DESCRIPTION
**Pull request by sunfishcode**
Update the WASI build to LLVM 14.0 and the wasi-libc version from wasi-sdk-15.

This will require a ci-mirrors.rust-lang.org file load. Specifically, we
need [this LLVM release tarball] uploaded to be downloadable from
[this URL].

[this LLVM release tarball]: https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz
[this URL]: https://ci-mirrors.rust-lang.org/rustc/2022-05-10-clang%2Bllvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz

### CODE REVIEW

File: src/ci/docker/host-x86_64/dist-various-2/build-wasi-toolchain.sh [hunk 1]
```diff
@@ -2,15 +2,15 @@
 
 set -ex
 
-# Originally from https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/clang+llvm-11.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz
-curl https://ci-mirrors.rust-lang.org/rustc/2021-01-14-clang%2Bllvm-11.0.1-x86_64-linux-gnu-ubuntu-16.04.tar.xz | \
+# Originally from https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+curl https://ci-mirrors.rust-lang.org/rustc/2022-05-10-clang%2Bllvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz | \
```
Comment (Mark-Simulacrum) [inline src/ci/docker/host-x86_64/dist-various-2/build-wasi-toolchain.sh:6]: It looks like dist-various-2 is using 20.04 Ubuntu now -- should we be downloading a 20.04 image here as well?

**Comment by Mark-Simulacrum** [comment]
I uploaded the file to our mirror bucket, so modulo comment r=me presuming 18.04 is OK.

File: src/ci/docker/host-x86_64/dist-various-2/build-wasi-toolchain.sh (hunk 1 above)
Comment (bjorn3) [reply src/ci/docker/host-x86_64/dist-various-2/build-wasi-toolchain.sh:6]: Only an ubuntu 18.04 tarball is provided by https://github.com/llvm/llvm-project/releases/tag/llvmorg-14.0.0 It should be fine though due to glibc being mostly backwards compatible.

**Comment by jonhoo** [comment]
I know I'm late to the game here, but I'm curious what the policy is on updating the specific commit of wasi-libc — does Rust give any guarantees about which version is used? If not, should it? Could totally be that the answer to both is "no", I just wanted to check :)

**Comment by sunfishcode** [comment]
I don't believe Rust currently makes any guarantees about the specific commit of wasi-libc. wasi-libc currently attempts to preserve ABI compatibility.

In the future, wasi-libc may want to make some ABI-incompatible changes. When we reach that point, we'll coordinate that with other toolchains and figure out how to commincate it with users.

**Pull request merged**